### PR TITLE
draft for Issue 54 models

### PR DESCRIPTION
**Pull request by martinjrobins**
An initial draft for #54. I think we'll have to work on the schema a bit, I just added the fields that were in the issue. But just to get it started

### CODE REVIEW

File: pkpdapp/pkpdapp/models/biomarker_type.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# is released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+
+from django.db import models
+
+
+class BiomarkerType(models.Model):
+    """
+    A biomarker type. Eache value, or biomarker, in a dataset must be one of
```
Comment (DavAug) [inline pkpdapp/pkpdapp/models/biomarker_type.py:12]: Same here.

**Review by DavAug — CHANGES_REQUESTED** [review]
Thanks @martinjrobins I think that looks really good!

I just have a couple minor comments on the documentation. 

So are we going to write tests for each migration now? Or are we just writing a test that checks that the after the migrations are performed the database is set up the way we would like it to be.

File: pkpdapp/pkpdapp/models/biomarker.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+# copyright notice and full license details.
+#
+
+from django.db import models
+from pkpdapp.models import BiomarkerType, Dataset
+
+
+class Biomarker(models.Model):
+    """
+    A biomarker measurement value for use in a dataset
```
[the remark this answers is not in the document]
Comment (martinjrobins) [reply pkpdapp/pkpdapp/models/biomarker.py:13]: yea, good point. I didn't write these cause I couldn't find the documentation on how to write docstrings for django, but now I have (https://docs.djangoproject.com/en/3.1/ref/contrib/admin/admindocs/). Looks like you put help strings for each field in the `help_text` attribute. I'll do that now

**Comment by martinjrobins** [comment]
> So are we going to write tests for each migration now? Or are we just writing a test that checks that the after the migrations are performed the database is set up the way we would like it to be.

I think we want to aim for testing at least the vast majority of all code, which includes the migrations. Note that I only wrote an explicit test for unapplying the 2nd migration, as this is otherwise not covered (as codecov told me :) ). I don't think we need to explicitly test applying the 1st and 2nd migration, as the tests of the models itself cover these (i.e. if the migrations failed, then the model tests will also fail).

**Review by DavAug — APPROVED** [review]
Thanks @martinjrobins . Quite cool that Django constructs the documentation automatically!

**Comment by ben18785** [comment]
@martinjrobins Do we need both this PR and #59? It looks like this one creates models which #59 also does...

**Comment by martinjrobins** [comment]
> @martinjrobins Do we need both this PR and #59? It looks like this one creates models which #59 also does...

This PR branch is branched from the #59 PR, as they follow on from each other. But they address different issues, so I kept them separate.

**Comment by ben18785** [comment]
@martinjrobins Ok, thanks -- sorry for duncing!

**Pull request merged**